### PR TITLE
Merge results with existing README.md

### PR DESCRIPTION
**Pull request by Samoed**
## Checklist

- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`. 

Merge new results with existing readme.

### CODE REVIEW

**Comment by Samoed** [comment]
I tried to add new scores for the `multilingual-e5-small` model, and everything seems to look fine. The model name lacked the repository owner, so I changed it from `multilingual-e5-small` to `intfloat/multilingual-e5-small`. The only issue is that the keys are shuffled, making the results look a bit messy in the diff because Python dictionaries don't preserve key order.

[original_readme.md](https://github.com/user-attachments/files/16404353/original_readme.md)
[new_readme.md](https://github.com/user-attachments/files/16404361/new_readme.md)

File: mteb/cli.py [hunk 1]
```diff
@@ -301,6 +302,56 @@ def potentially_add_cqadupstack_to_results(results: list[mteb.MTEBResults]) -> N
     results.append(result)
 
 
+def merge_yamls(
+    yaml_dict: Dict[str, Any], existing_readme: str
+) -> Tuple[Dict[str, Any], str]:
+    if not existing_readme.endswith("md"):
+        return yaml_dict, ""
```
Comment (KennethEnevoldsen) [inline mteb/cli.py:309]: I would raise an error instead

File: mteb/cli.py [hunk 2]
```diff
@@ -301,6 +302,56 @@ def potentially_add_cqadupstack_to_results(results: list[mteb.MTEBResults]) -> N
...
+
+    with open(existing_readme) as f:
```
Comment (KennethEnevoldsen) [inline mteb/cli.py:311]: What happens if the file does not have an frontmatter? Should we add in an error for that

**Review by KennethEnevoldsen — APPROVED** [review]
Thanks!! This looks great. Only a few minor comments. Will you mind updating the the docs to match as well:

https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_model.md

**Review by KennethEnevoldsen — APPROVED** [review]

**Pull request merged**